app/celery/process_delivery_status_result_tasks.py

```diff
@@ -1,10 +1,224 @@
-from flask import current_app
-from app import notify_celery
+import math
+import json
+import time
+import datetime
+from app.celery.service_callback_tasks import check_and_queue_callback_task
 from app.celery.process_pinpoint_inbound_sms import CeleryEvent
 
+from app.dao.notifications_dao import (
+    dao_get_notification_by_reference,
+    dao_update_notification,
+    update_notification_status_by_id,
+)
+
+from typing import Tuple
+from celery.exceptions import Retry
+from flask import current_app
+# import the clients instance from the app
+from notifications_utils.statsd_decorators import statsd
+from sqlalchemy.orm.exc import NoResultFound, MultipleResultsFound
+from app import notify_celery, statsd_client, clients, DATETIME_FORMAT
+from app.config import QueueNames
+from app.dao.service_callback_dao import dao_get_callback_include_payload_status
+from app.feature_flags import FeatureFlag, is_feature_enabled
+
+from app.models import (
+    NOTIFICATION_DELIVERED,
+    NOTIFICATION_TECHNICAL_FAILURE,
+    NOTIFICATION_PERMANENT_FAILURE,
+    Notification, NOTIFICATION_PREFERENCES_DECLINED
+)
+FINAL_STATUS_STATES = [NOTIFICATION_DELIVERED, NOTIFICATION_PERMANENT_FAILURE, NOTIFICATION_TECHNICAL_FAILURE,
+                       NOTIFICATION_PREFERENCES_DECLINED]
+
 
 # Create SQS Queue for Process Deliver Status.
 @notify_celery.task(bind=True, name="process-delivery-status-result", max_retries=48, default_retry_delay=300)
+@statsd(namespace="tasks")
 def process_delivery_status(self, event: CeleryEvent) -> bool:
+    """ Celery task for updating the delivery status of a notification """
+
+    # preset variables to address "unbounded local variable"
+    sqs_message = None
+    notification_platform_status = None
+
+    if not is_feature_enabled(FeatureFlag.PROCESS_DELIVERY_STATUS_ENABLED):
+        current_app.logger.info('Process Delivery Status toggle is disabled.  Skipping callback task.')
+        return True
+
+    # log that we are processing the delivery status
     current_app.logger.info('processing delivery status: %s', event)
+
+    # first attempt to process the incoming event
+    try:
+        sqs_message = json.loads(event['Message'])
+    except (json.decoder.JSONDecodeError, ValueError, TypeError, KeyError) as e:
+        current_app.logger.exception(e)
+        self.retry(queue=QueueNames.RETRY)
```

Comment:
I don't see any point to retrying a mangled event because it will always be a manged event.

@k-macmillan @jakehova I think we discussed this elsewhere, but I don't remember if we changed the behavior.  What do you think?